editor/editor_interface.cpp

```diff
@@ -512,6 +513,39 @@ void EditorInterface::popup_quick_open(const Callable &p_callback, const TypedAr
 	quick_open->popup_dialog(base_types, callable_mp(this, &EditorInterface::_quick_open).bind(p_callback));
 }
 
+void EditorInterface::popup_create_dialog(const Callable &p_callback, const bool p_replace_mode, const StringName &p_base_type, const String &p_current_type_name, const String &p_replace_type_name, const TypedArray<StringName> &p_custom_type_blacklist, const Dictionary &p_custom_suffix) {
+	if (!create_dialog) {
+		create_dialog = memnew(CreateDialog);
+		get_base_control()->add_child(create_dialog);
+	}
+
+	HashSet<StringName> blacklist;
+	for (const Variant &E : p_custom_type_blacklist) {
+		blacklist.insert(E);
+	}
+	create_dialog->set_type_blacklist(blacklist);
+
+	HashMap<StringName, String> suffix_map;
+	List<Variant> keys;
+	p_custom_suffix.get_key_list(&keys);
+	for (Variant k : keys) {
+		StringName key = k.operator StringName();
+		if (key.is_empty()) {
+			// Skips empty keys and invalid values. The dictionary should be Dictionary[StringName, String],
+			// but due to a bug when using TypedDictionary, the final compile will failed due to being unable to
+			// parse ";" after "typeddictionary::" during the godot-cpp binding.
```

Comment:
This note isn't needed IMO, instead there should probably be a note on the method to consider using a typed dictionary